docs/guides/multisig-proposal.md

```diff
@@ -0,0 +1,241 @@
+        address token = addresses.getAddress("TOKEN_1");
+        _pushAction(
+            timelockVault,
+            abi.encodeWithSignature(
+                "whitelistToken(address,bool)",
+                token,
+                true
+            ),
+            "Set token to active"
+        );
+    }
+
+    // Executes the proposal actions.
+    function _run(Addresses addresses, address) internal override {
+        address multisig = addresses.getAddress("DEV_MULTISIG");
+
+        _simulateActions(multisig);
+    }
+
+    // Validates the post-execution state.
+    function _validate(Addresses addresses, address) internal override {
+        address devMultisig = addresses.getAddress("DEV_MULTISIG");
+        Vault timelockVault = Vault(addresses.getAddress("VAULT"));
+        MockToken token = MockToken(addresses.getAddress("TOKEN_1"));
+
+        assertEq(timelockVault.owner(), devMultisig);
+        assertTrue(timelockVault.tokenWhitelist(address(token)));
+        assertFalse(timelockVault.paused());
+
+        assertEq(token.owner(), devMultisig);
+        assertEq(token.balanceOf(devMultisig), token.totalSupply());
+    }
+}
+```
+
+Let's go through each of the functions we are overriding here.
+
+-   `name()`: Define the name of your proposal.
+-   `description()`: Provide a detailed description of your proposal.
+-   `_deploy()`: Deploy any necessary contracts. This example demonstrates the deployment of Vault and an ERC20 token.
+-   `_build(`): Set the necessary actions for your proposal. In this example, ERC20 token is whitelisted on the Vault contract
+-   `_run()`: Execute the proposal actions outlined in the `_build()` step. This
+    function performs a call to `simulateActions()` from the inherited
+    `MultisigProposal` contract. Internally, `_simulateActions()` simulates a call to the [Multicall3](https://www.multicall3.com/) contract with the calldata generated from the actions set up in the build step.
+-   `_validate()`: This final step is crucial for validating the post-execution state. It ensures that the multisig is the new owner of Vault and token, the tokens were transferred to multisig and the token was whitelisted on the Vault contract
+
+Now that your first proposal contract is ready, it's time to take action. You have two options to execute the contract. The first option is to use `foundry test`. You can learn how to do that on [integration-tests.md](../testing/integration-tests.md "mention") section. The second option is to use `foundry script`, which is the method we will use here. But first, we need to set up [Addresses](../overview/architecture/addresses.md) contract. Let's create a `Addresses.json` file:
+
+```json
+[
+    {
+        "addr": "0x3dd46846eed8D147841AE162C8425c08BD8E1b41",
+        "name": "DEV_MULTISIG",
+        "chainId": 31337
+    }
+]
+```
+
+Now that we have the JSON file to be use on `Addresses.sol`, let's create a script that inherits `ScriptSuite`.
+
+```solidity
```

Comment:
What do we think about moving these code examples into the `examples/` directory?